lib/ExpensiMark.js

```diff
@@ -476,9 +470,28 @@ export default class ExpensiMark {
             }
             replacedText = replacedText.concat(textToCheck.substr(startIndex, (match.index - startIndex)));
 
+            // we want to avoid matching email address domains
+            let abort = false;
+            let shouldRetryByAtSign = false;
+
+            if ((match.index !== 0) && (textToCheck[match.index - 1] === '@')) {
+                const domainRegex = new RegExp('^(([a-z-0-9]+\\.)+[a-z]{2,})(\\S*)', 'i');
+                const domainMatch = domainRegex.exec(match[2]);
+
+                // e.g. test@expensify.com/https://www.test.com
+                // If the matched string faces @ sign before it,
+                // We will retry to apply autolink rule to the string(e.g. /https://www.test.com) except for domain(e.g. expensify.com) after @ sign.
+                if ((domainMatch !== null) && (domainMatch[3] !== '')) {
+                    shouldRetryByAtSign = true;
+                    replacedText = replacedText.concat(domainMatch[1] + this.replace(domainMatch[3], {filterRules: ['autolink']}));
+                } else {
+                    abort = true;
```

Comment:
Would it make sense to add a comment explaining why we're setting abort to true here? It's not immediately obvious to me

Reply:
![image](https://github.com/Expensify/expensify-common/assets/139204153/0a59cc97-3bac-4dbd-b4af-abc9872a01e0)

This comment is to say why we are setting abort to true